### PR TITLE
Code refactoring for unit testing

### PR DESCRIPTION
**Pull request by benjaminbillet**
Co-Authored-By Abderraouf BEN SALEM <Abderraouf.BENSALEM@amadeus.com>

### CODE REVIEW

**Comment by benjaminbillet** [comment]
Hello,
A little bit of context for this pull request:

- create a class containing configuration options (AppConfig)
- decouple various interactions (the kafka operator was referenced in a lot of places)
- split the KafkaUtilities big class into smaller classes: KafkaAdmin for low-level topic management, TopicManager for the operator-specific topic-management logic
- splitting the concerns of ConfigMapManager (it was used both for watching and creating config maps) into two hierarchy: the "watch" package, containing classes that watch resources, and the "importer" package, that implements the topic import logic and the subsequent creation of resources
- introduce an .editorconfig file (https://EditorConfig.org)
- introduce basic unit tests, using junit and mockito

Best regards,
Benjamin Billet

File: src/main/java/nb/kafka/operator/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  public void setSecurityProtocol(String securityProtocol) {
+    this.securityProtocol = securityProtocol;
+  }
+  public short getDefaultReplicationFactor() {
+    return defaultReplicationFactor;
+  }
+  public void setDefaultReplicationFactor(short defaultReplicationFactor) {
+    this.defaultReplicationFactor = defaultReplicationFactor;
+  }
+  public boolean isEnableTopicDelete() {
```
Comment (nbogojevic) [inline src/main/java/nb/kafka/operator/AppConfig.java]: I'd prefer isEnabledTopicDelete to be more in spirit of English - also setEnabled...

File: src/test/java/nb/kafka/operator/util/FieldUtil.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package nb.kafka.operator.util;
+
+import java.lang.reflect.Field;
+
+public final class FieldUtil {
```
Comment (nbogojevic) [inline src/test/java/nb/kafka/operator/util/FieldUtil.java]: I couldn't find where is this used?

Comment (benjaminbillet) [reply src/test/java/nb/kafka/operator/util/FieldUtil.java]: Ah yes, we were planning to write more tests but ran out of time. This helper will be used later ; I'll remove it from this PR

File: src/main/java/nb/kafka/operator/AppConfig.java (hunk 1 above)
Comment (benjaminbillet) [reply src/main/java/nb/kafka/operator/AppConfig.java]: The Eclipse getter/setter generator is not very english-compliant :)
I'll change that.

**Pull request merged**